juju/juju
# feat(sql): STRICT sql tables

SQLite by default uses dynamic untyped columns. There is no validation of columns on the way in or out. The proposal is to enable strict tables which will help with locating potential failures with DDL and statement queries.

All DATETIME and TIMESTAMP columns become TEXT. Then BOOLEAN becomes INT (represented as 0 or 1).



## Checklist




## QA steps

This hopefully shouldn't cause any issues, but in any case, a set of regression steps should be done:

```
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```

## Links


**Jira card:** [JUJU-6163](https://warthogs.atlassian.net/browse/JUJU-6163)



[JUJU-6163]: https://warthogs.atlassian.net/browse/JUJU-6163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
